app/controllers/two_factor_authentication/otp_verification_controller.rb

```diff
@@ -0,0 +1,19 @@
+module TwoFactorAuthentication
+  class OtpVerificationController < DeviseController
+    include TwoFactorAuthenticatable
+
+    def show
+      @phone_number = user_decorator.masked_two_factor_phone_number
+      @code_value = current_user.direct_otp if FeatureManagement.prefill_otp_codes?
+      @delivery_method = request.fullpath.split('/').last
```

Comment:
Excellent point. I failed to apply the the open/closed principle by duplicating the routes instead of making them dynamic. Check out my latest commit that fixes this.